### ISSUE TITLE
Use Netty as a backend instead of java.io for much more efficient networking.

### ISSUE THREAD

**Issue by me4502** [BlockServerProject/JRakLib#1]
Netty handles threading very well, unlike java.io sockets. Currently there is a large bottleneck in terms of single CPU performance due to usage of these old APIs.

**Comment by alejzeis** [comment]
Yes, I was thinking of using Netty, but I am not that well familiar with
it. If you could help, that would be great (on a seperate branch first).

On Wednesday, August 5, 2015, Matthew Miller notifications@github.com
wrote:

> Netty handles threading very well, unlike java.io sockets. Currently
> there is a large bottleneck in terms of single CPU performance due to usage
> of these old APIs.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/BlockServerProject/JRakLib/issues/1.

##

**Comment by net32** [comment]
I am finishing my version of JRakLib fully using ionetty, I'll publish it here when committing, not using while (true), is in the final version already, I'm just packing up some minor bugs, performance is very good, very low CPU usage.

**Comment by net32** [comment]
@jython234 you could create a new branch, named ioNetty ? thanks

**Comment by alejzeis** [comment]
@net32 Done.

**Comment by net32** [comment]
@me4502 see: https://github.com/net32/JRakLib/tree/ioNetty

**Comment by RazvanDu** [comment]
Does the netty branch really improve performance?

**Comment by alejzeis** [comment]
I tested it and it actually appeared to be slower than master.

**Comment by me4502** [comment]
If used correctly, Netty should provide better performance.
